quickwit/quickwit-indexing/src/source/file_source.rs

```diff
@@ -121,35 +160,51 @@ impl TypedSourceFactory for FileSourceFactory {
         source_runtime: SourceRuntime,
         params: FileSourceParams,
     ) -> anyhow::Result<FileSource> {
-        let Some(uri) = &params.filepath else {
-            return Ok(FileSource {
-                source_id: source_runtime.source_id().to_string(),
-                reader: DocFileReader::from_stdin(),
-                partition_id: None,
-                counters: FileSourceCounters::default(),
-            });
+        let source_id = source_runtime.source_config.source_id.clone();
+        let source_type = source_runtime.source_config.source_type();
+        let inner = match params {
+            FileSourceParams::FileUri(file_uri) => {
+                let partition_id = PartitionId::from(file_uri.filepath.as_str());
+                let position = source_runtime
+                    .fetch_checkpoint()
+                    .await?
+                    .position_for_partition(&partition_id)
+                    .cloned()
+                    .unwrap_or_default();
+                let batch_reader = ObjectUriBatchReader::try_new(
+                    &source_runtime.storage_resolver,
+                    partition_id,
+                    &file_uri.filepath,
+                    position,
+                )
+                .await?;
+                FileSourceInner::Stream {
+                    reader: Box::new(batch_reader),
+                    num_bytes_processed: 0,
+                    num_lines_processed: 0,
+                }
+            }
+            FileSourceParams::Stdin => FileSourceInner::Stream {
+                reader: Box::new(StdinBatchReader::new()),
+                num_bytes_processed: 0,
+                num_lines_processed: 0,
+            },
+            #[cfg(feature = "sqs")]
+            FileSourceParams::Sqs(sqs_config) => {
```

Comment:
for pubsub what will to look like?

Reply:
https://github.com/quickwit-oss/quickwit/pull/5148#discussion_r1662158060